src/form-engine.component.tsx

```diff
@@ -107,6 +109,12 @@ const FormEngine: React.FC<FormProps> = ({
   const postSubmissionHandlers = usePostSubmissionAction(refinedFormJson?.postSubmissionActions);
   const sessionMode = mode ? mode : encounterUUID || encounterUuid ? 'edit' : 'enter';
   const { isFormExpanded, hideFormCollapseToggle } = useFormCollapse(sessionMode);
+  const [hasPatientWorkflows, setHasPatientWorkflows] = useState(false);
+  const {
+    isLoading: isLoadingPatientPrograms,
+    error: patientProgramsError,
+    patientEnrollment,
+  } = usePatientEnrollment(patientUUID);
```

Comment:
Two concerns here:

1. Unless you have a good reason, move this logic to the encounter component
2. Is it necessary to load enrolments for scenarios where the form doesn't have any related fields?

Reply:
I can move the logic to the encounter component